python/GafferDispatchUI/FrameMaskUI.py

```diff
@@ -0,0 +1,69 @@
+##########################################################################
+#
+#  Copyright (c) 2018, Image Engine Design Inc. All rights reserved.
+#
+#  Redistribution and use in source and binary forms, with or without
+#  modification, are permitted provided that the following conditions are
+#  met:
+#
+#      * Redistributions of source code must retain the above
+#        copyright notice, this list of conditions and the following
+#        disclaimer.
+#
+#      * Redistributions in binary form must reproduce the above
+#        copyright notice, this list of conditions and the following
+#        disclaimer in the documentation and/or other materials provided with
+#        the distribution.
+#
+#      * Neither the name of John Haddon nor the names of
+#        any other contributors to this software may be used to endorse or
+#        promote products derived from this software without specific prior
+#        written permission.
+#
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+#  IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+#  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+#  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+#  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+#  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+#  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+#  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+#  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+#  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+#  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+##########################################################################
+
+import Gaffer
+import GafferUI
+import GafferDispatch
+
+Gaffer.Metadata.registerNode(
+
+	GafferDispatch.FrameMask,
+
+	"description",
+	"""
+	Masks upstream tasks so that they will only be executed for
+	a subset of the Dispatcher's frame range.
+	""",
+
+	plugs = {
+
+		"mask" : (
+
+			"description",
+			"""
+			The subset of frames that will be executed by upstream tasks.
+			Any frames not included here will be ignored, regardless
+			of the dispatcher's frame range.
+
+			> Note : This can only remove frames. To add frames, edit the
```

Comment:
A TaskTimeWarp wouldn't add frames either - it would offset or multiply them. It's not impossible to add frames, but you'd have to declare them as preTasks of something, which would make for pretty odd looking job structures in something like Tractor. A FrameMask is a simple concept intended for a specific purpose that has come up multiple times, and I don't think additional features are needed.

If we truly do want to add frames (and that's not what was asked for here), then I think the way to do it is to have some sort of auto-discovery mechanism whereby the Dispatcher gets the valid frame ranges automatically by examining the TaskNodes. But even then, what would adding frames achieve? If an upstream task says it is only valid for a particular range, then logically we can only mask from that range anyway, not add to it.